ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala

```diff
@@ -0,0 +1,159 @@
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import com.digitalasset.daml.lf.data.Ref
+import com.digitalasset.daml.lf.data.Time.Timestamp
+import com.digitalasset.daml.lf.transaction.{BlindingInfo, GenTransaction}
+import com.digitalasset.daml.lf.value.Value
+import com.digitalasset.daml.lf.value.Value.{AbsoluteContractId, NodeId}
+import com.digitalasset.ledger.api.domain._
+
+package object v2 {
+
+  final case class AcsUpdate(
+      optSubmitterInfo: Option[SubmitterInfo],
+      offset: LedgerOffset,
+      transactionMeta: TransactionMeta,
+      transactionId: TransactionId,
+      events: List[AcsUpdateEvent]
+  )
+
+  sealed trait AcsUpdateEvent extends Product with Serializable
+
+  object AcsUpdateEvent {
+
+    final case class Create(
+        transactionId: TransactionId,
+        nodeId: NodeId,
+        contractId: Value.AbsoluteContractId,
+        templateId: Ref.Identifier,
+        argument: Value.VersionedValue[Value.AbsoluteContractId],
+        // TODO(JM,SM): understand witnessing parties
+        stakeholders: List[Ref.Party],
+    ) extends AcsUpdateEvent
+
+    final case class Archive(
+        transactionId: TransactionId,
+        nodeId: NodeId,
+        contractId: Value.AbsoluteContractId,
+        templateId: Ref.Identifier,
+        // TODO(JM,SM): understand witnessing parties
+        stakeholders: List[Ref.Party],
+    ) extends AcsUpdateEvent
+
+  }
+
+  sealed trait CompletionEvent extends Product with Serializable {
+    def offset: LedgerOffset.Absolute
+  }
+
+  object CompletionEvent {
+
+    final case class Checkpoint(offset: LedgerOffset.Absolute, recordTime: Timestamp)
+        extends CompletionEvent
+
+    final case class CommandAccepted(
+        offset: LedgerOffset.Absolute,
+        commandId: CommandId,
+        transactionId: TransactionId)
+        extends CompletionEvent
+
+    final case class CommandRejected(
+        offset: LedgerOffset.Absolute,
+        commandId: CommandId,
+        reason: RejectionReason)
+        extends CompletionEvent
+
+  }
+
+  final case class ActiveContractSetSnapshot(
+      takenAt: LedgerOffset.Absolute,
+      activeContracts: Source[(WorkflowId, AcsUpdateEvent.Create), NotUsed])
+
+  /** A transaction that has been accepted as committed by the Participant
+    * node.
+    *
+    * @param transactionData: the transaction that was accepted as committed.
+    *
+    * @param transactionMeta: Meta-data of a transaction visible to all parties
+    *                       that can see a part of the transaction.
+    *
+    * @param submitterInfo: information about the original submission of the transaction. Is [[None]]
+    *   if the participant node does not host the submitter.
+    *
+    * @param blindingInfo: Information for divulgence and disclosure.
+    *
+    */
+  final case class Transaction(
+      transactionData: GenTransaction.WithTxValue[NodeId, AbsoluteContractId],
+      blindingInfo: BlindingInfo,
```

Comment:
Should not be needed. The filtering that needs this should happen within the index service.